compiler/rustc_codegen_llvm/src/builder.rs

```diff
@@ -769,7 +769,7 @@ impl BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
         // we like. To ensure that LLVM picks the right instruction we choose
         // the raw wasm intrinsic functions which avoid LLVM inserting all the
         // other control flow automatically.
-        if self.sess().target.arch == "wasm32" {
+        if self.sess().target.arch.starts_with("wasm32") {
```

Comment:
```suggestion
        if self.sess().target.arch.starts_with("wasm") {
```
or i'm pretty sure this exists:
```suggestion
		if self.sess().target.is_like_wasm {
```